src/main/kotlin/no/skatteetaten/aurora/mokey/service/ApplicationDataServiceOpenShift.kt

```diff
@@ -31,29 +31,22 @@ class ApplicationDataServiceOpenShift(
     val addressService: AddressService,
     val imageService: ImageService
 ) : ApplicationDataService {
-    override fun findAllVisibleAffiliations(): List<String> {
-        throw NotImplementedError("findAllVisibleAffiliations is not supported")
-    }
 
     val mtContext = newFixedThreadPoolContext(6, "mokeyPool")
 
     val logger: Logger = LoggerFactory.getLogger(ApplicationDataServiceOpenShift::class.java)
 
     override fun findAllAffiliations(): List<String> {
-        return findAllEnvironments().map { it.affiliation }.toSet().toList()
+        return findAndGroupAffiliations().keys.toList()
     }
 
     override fun findAllApplicationData(affiliations: List<String>, ids: List<String>): List<ApplicationData> {
-        logger.debug("finding application for affiliations=$affiliations")
-        val apps = if (affiliations.isEmpty()) {
-            logger.debug("finding applications in all envs")
-            findAllApplicationDataByEnvironments()
-        } else {
-            val allEnvironments = findAllEnvironments()
-            val environmentsForAffiliations = allEnvironments.filter { affiliations.contains(it.affiliation) }
-            findAllApplicationDataByEnvironments(environmentsForAffiliations)
-        }
-        return apps.filter { if (ids.isEmpty()) true else ids.contains(it.applicationDeploymentId) }
+        val affiliationGroups = findAndGroupAffiliations(affiliations)
+        return findAllApplicationDataForEnv(ids, affiliationGroups)
+    }
+
+    override fun findAllVisibleAffiliations(): List<String> {
```

Comment:
This is a general comment about the design of the decorator, and not specifically the changes in the PR.
I feel we are moving in a direction here that I do not like. The implementations of `ApplicationDataService` are becoming more and more different, even providing separate functions. For me this should be split into two separate classes at this point, for me the way it is now is just confusing to have this as a decorator.

Reply:
I think we should discuss this at a Team meeting.